src/Purifier/OrderBonusPointsPurifier.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusBonusPointsPlugin\Purifier;
+
+use BitBag\SyliusBonusPointsPlugin\Context\CustomerBonusPointsContextInterface;
+use BitBag\SyliusBonusPointsPlugin\Entity\AdjustmentInterface;
+use BitBag\SyliusBonusPointsPlugin\Entity\BonusPointsInterface;
+use Doctrine\Persistence\ObjectManager;
+
+final class OrderBonusPointsPurifier implements OrderBonusPointsPurifierInterface
+{
+    /** @var CustomerBonusPointsContextInterface */
+    private $customerBonusPointsContext;
+
+    /** @var ObjectManager */
+    private $manager;
+
+    public function __construct(
+        CustomerBonusPointsContextInterface $customerBonusPointsContext,
+        ObjectManager $manager
+    ) {
+        $this->customerBonusPointsContext = $customerBonusPointsContext;
+        $this->manager = $manager;
+    }
+
+    public function purify(BonusPointsInterface $bonusPoints): void
+    {
+        $order = $bonusPoints->getOrder();
+        $customerBonusPoints = $this->customerBonusPointsContext->getCustomerBonusPoints();
+
+        if ($order->getCustomer() !== $customerBonusPoints->getCustomer()) {
+            return;
+        }
+
+        $order->removeAdjustmentsRecursively(AdjustmentInterface::ORDER_BONUS_POINTS_ADJUSTMENT);
+        $customerBonusPoints->removeBonusPointsUsed($bonusPoints);
+        $bonusPoints->setPoints(0);
+
+        $this->manager->persist($bonusPoints);
+        $this->manager->persist($customerBonusPoints);
+        $this->manager->persist($order);
+        $this->manager->flush();
```

Comment:
I'm not a fan for flushing entities in service. The persistence flushing should be done at the end of entire process, so it's responsibility for handlers/actions. Is there any reason to flush it here? Could we flush it outside service?